### PR TITLE
feat(versions): Add version group headers based on relative date

### PR DESCRIPTION
**Pull request by jstoffan**
grouping

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

File: src/elements/content-sidebar/versions/VersionsList.js [hunk 1]
```diff
@@ -34,20 +35,36 @@ const VersionsList = ({ isLoading, versions, ...rest }: Props) => {
...
+                const isCurrent = index === 0;
+                const previousVersion = versions[Math.max(0, index - 1)];
+                const previousGroup = getGroup(previousVersion);
+                const newGroup = getGroup(version);
+                const hasGroup = isCurrent || previousGroup !== newGroup;
+
+                return (
+                    <React.Fragment key={version.id}>
+                        {hasGroup && (
+                            <li className="bcs-VersionsList-heading">
```
Comment (ConradJChan) [inline src/elements/content-sidebar/versions/VersionsList.js]: Is there another semantic way to do subheaders in a list? I'm wondering how screen readers might read this

File: src/elements/content-sidebar/versions/VersionsList.scss [hunk 2]
```diff
@@ -11,6 +11,17 @@
...
 }
 
+.bcs-VersionsList-heading {
+    color: $bdl-gray-50;
+    font-size: 14px;
+    padding-bottom: 10px;
+    padding-top: 15px;
+
+    &:empty {
+        display: none;
```
Comment (ConradJChan) [inline src/elements/content-sidebar/versions/VersionsList.scss]: When would this occur?

Comment (jstoffan) [reply src/elements/content-sidebar/versions/VersionsList.scss]: It won't; it's a stray declaration from an earlier revision. Good catch, will remove.

File: src/elements/content-sidebar/versions/VersionsList.js (hunk 1 above)
Comment (jstoffan) [reply src/elements/content-sidebar/versions/VersionsList.js]: Good question. This is how our dropdown menu works, so I followed the convention here as well. I'll take a quick look online to see what I find.

File: src/elements/content-sidebar/versions/VersionsSidebar.js [hunk 3]
```diff
@@ -10,42 +10,71 @@ import InlineError from '../../../components/inline-error';
...
+
+                {showEmpty && (
+                    <div className="bcs-Versions-empty">
+                        <FormattedMessage {...messages.versionsEmpty} />
+                    </div>
+                )}
+
+                {showVersions && (
+                    <ul className="bcs-Versions-menu">
+                        {Object.keys(versionGroups).map(versionGroupKey => (
```
Comment (ConradJChan) [inline src/elements/content-sidebar/versions/VersionsSidebar.js]: Do we need to worry about sorting the order of the versionGroups? Assuming the API data comes back in a sorted order this should work, but I'm thinking about the corner case of if version has no `createdAt` (is that even possible?) in which case it gets no heading and has a key of `''` and I don't know where that would appear in the `Object.keys` order

Comment (jstoffan) [reply src/elements/content-sidebar/versions/VersionsSidebar.js]: It doesn't look like `created_at` is a nullable field and the server explicitly sorts versions by that value for its response. I've removed the unnecessary existence checks from the new components.

**Pull request merged**